Course/Course/Presentation/Subviews/CustomDisclosureGroup.swift

```diff
@@ -218,9 +216,32 @@ struct CustomDisclosureGroup: View {
         }
     }
     
+    private func assignmentStatusText(
+        sequential: CourseSequential
+    ) -> String? {
+        
+        guard let sequentialProgress = sequential.sequentialProgress,
+              let assignmentType = sequentialProgress.assignmentType,
+              let numPointsEarned = sequentialProgress.numPointsEarned,
+              let numPointsPossible = sequentialProgress.numPointsPossible,
+              let due = sequential.due else {
+            return nil
+        }
+        
+        let daysRemaining = getAssignmentStatus(for: due)
+        
+        if let courseVertical = sequential.childs.first,
+           courseVertical.childs.isEmpty {
+            return "\(assignmentType) - \(daysRemaining)"
+        }
+        
+        return "\(assignmentType) - \(daysRemaining) - \(numPointsEarned) / \(numPointsPossible)"
+        
+    }
+    
     private func canDownloadAllSections(in chapter: CourseChapter) -> Bool {
         for sequential in chapter.childs {
-            if let state = viewModel.sequentialsDownloadState[sequential.id] {
+            if let _ = viewModel.sequentialsDownloadState[sequential.id] {
```

Comment:
If we refactor it something like this:
```
    private func canDownloadAllSections(in chapter: CourseChapter) -> Bool {
        for sequential in chapter.childs where viewModel.sequentialsDownloadState[sequential.id] != nil {
            return true
        }
        return false
    }
```

Then we can avoid following two warnings:
```
Unused Optional Binding Violation: Prefer `!= nil` over `let _ =` (unused_optional_binding)

Prefer For-Where Violation: `where` clauses are preferred over a single `if` inside a `for` (for_where)
```